### PR TITLE
Network utilities base class

### PR DESCRIPTION
**Pull request by srmoore**
The network utilities have been made into a class. `device.utilities.network.base_network_utility` defines an abstract `NetworkUtility` class. There are two implementations of this class: `GenericNetworkUtility` contains much of the old system of scripts and should work on BBBw and running off a laptop/etc. `BalenaNetworkUtility` is a specific instance that is used in Balena environments to trigger dropping off the wifi-network, and firing up balena.io's wifi-connect captive portal access point for configuring a new network connection. 

The new system works on Balena. Should be tested on a laptop and probably a BBB without Balena before merging

### CODE REVIEW

File: device/utilities/network/base_network_utility.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import socket
+import urllib.request
+from abc import ABC, abstractmethod
+
+from typing import List, Dict
+
+from device.utilities.logger import Logger
+
+
+class NetworkUtility(ABC):
```
Comment (rbaynes) [inline device/utilities/network/base_network_utility.py]: Nice!

File: device/utilities/network/generic_network_utility.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
+import os, subprocess
+from typing import List, Dict
+
+from device.utilities.network.base_network_utility import NetworkUtility
+
+
+class GenericNetworkUtility(NetworkUtility):
+    def __init__(self) -> None:
+        super(GenericNetworkUtility, self).__init__()
+        self.GET_WIFI_SSIDS_SCRIPT_PATH = "scripts/network/get_wifi_ssids.sh"
```
Comment (rbaynes) [inline device/utilities/network/generic_network_utility.py]: Is this for backwards compatibility and/or OSX support?    If so, then its fine.   But I was thinking we get rid of those scripts for all platforms and use specialized classes that do what is in the scripts using subprocess.run()

**Comment by rbaynes** [comment]
I will test it on OSX now.

File: device/utilities/network/generic_network_utility.py (hunk 2 above)
Comment (srmoore) [reply device/utilities/network/generic_network_utility.py]: Exactly. I needed a place to put what we had until we replaced it

**Comment by rbaynes** [comment]
Looks OK on OSX, this is the only error I found:
[29/Mar/2019 19:46:24] "GET /api/iot/info/?csrfmiddlewaretoken=dLf8AQcnKH5xao67ykhCdgvDtsjhs6Mr5sS2latyQtRe3T2dulTp3UEHL5y3sfno HTTP/1.1" 200 112
date: illegal option -- -
usage: date [-jnRu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ... 
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
psql: FATAL:  role "openag_brain" does not exist

**Comment by rbaynes** [comment]
@srmoore  I'll let you delete the branch when you are done with it.  Thanks!

**Pull request merged**